chrome/content/zotero/xpcom/data/collection.js

```diff
@@ -23,7 +23,7 @@
     ***** END LICENSE BLOCK *****
 */
 
-Zotero.Collection = function() {
+Zotero.Collection = function(name) {
```

Comment:
The problem with this is that it's messier and less clear later if we add a required parameter. And setting the name before the libraryID is awkward. In the absence of named parameters, I think I'd prefer to just use properties. (We do use a required parameter for itemTypeID for Zotero.Item, but that one isn't ever going to change, it does have to be set before most other fields, and seems less weird before libraryID because it's actually defining the kind of item you're creating.)